src/main/java/eu/europeana/metis/sandbox/service/workflow/HarvestServiceImpl.java

```diff
@@ -0,0 +1,68 @@
+package eu.europeana.metis.sandbox.service.workflow;
+
+import eu.europeana.metis.harvesting.HarvesterException;
+import eu.europeana.metis.harvesting.ReportingIteration;
+import eu.europeana.metis.harvesting.http.CompressedFileExtension;
+import eu.europeana.metis.harvesting.http.HttpHarvester;
+import eu.europeana.metis.harvesting.http.HttpHarvesterImpl;
+import eu.europeana.metis.harvesting.http.HttpRecordIterator;
+import eu.europeana.metis.sandbox.common.exception.InvalidZipFileException;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.List;
+
+
+@Service
+public class HarvestServiceImpl implements HarvestService {
+
+
+    @Override
+    public List<ByteArrayInputStream> harvest(MultipartFile file) {
+        List<ByteArrayInputStream> records = new ArrayList<>();
+        HttpHarvester harvester = new HttpHarvesterImpl();
+        try {
+            harvester.harvestRecords(file.getInputStream(), CompressedFileExtension.ZIP, entry -> {
+                final byte[] content = entry.getEntryContent().readAllBytes();
+                records.add(new ByteArrayInputStream(content));
+            });
+
+        } catch (IOException | HarvesterException ex) {
+            throw new InvalidZipFileException(ex);
+        }
+
+        if (records.isEmpty()) {
+            throw new IllegalArgumentException("Provided file does not contain any records");
+        }
+
+        return records;
+    }
+
+    @Override
+    public List<ByteArrayInputStream> harvest(String URL) {
+
+        String tmpFolder = System.getProperty("java.io.tmpdir");
+        List<ByteArrayInputStream> records = new ArrayList<>();
+        HttpHarvester harvester = new HttpHarvesterImpl();
+        try {
+            HttpRecordIterator iterator = harvester.harvestRecords(URL, tmpFolder);
```

Comment:
I just noticed a resource leak.
This `harvesterRecords` method does not seem to clean the temporary directory/file that was downloaded/created(as compared to the other `harvestRecords`), which makes sense because it returns an iterator.
That means that somehow we need to clean those resources..